vortex-array/src/array/chunked/compute/filter.rs

```diff
@@ -106,24 +102,26 @@ fn filter_slices<'a>(
             }
         } else {
             // start != end means that the range is split over at least two chunks:
-            // start chunk: append a slice from (start_idx, start_chunk_end).
+            // start chunk: append a slice from (start_idx, start_chunk_end), i.e. whole chunk.
             // end chunk: append a slice from (0, end_idx).
             // chunks between start and end: append ChunkFilter::All.
-            let start_chunk_end = chunk_ends[start_chunk + 1];
-            let start_slice = (start_idx, start_chunk_end as _);
+            let start_chunk_len = chunk_ends[start_chunk + 1] - chunk_ends[start_chunk];
```

Comment:
This is the real bug - you have to append (start_idx, chunk_len), i.e. the rest of the chunk, ends are cumulative sums so you need to take the diff

Reply:
doh